*: upgrade yaml.v2 uses to v4

RaduBerinde opened the pull request:
#### go.mod: upgrade to yaml v4

The https://github.com/go-yaml/yaml (import path `gopkg.in/yaml.v3`)
repository has been archived. The YAML team took over maintenance in a
separate fork at a very confusingly named
https://github.com/yaml/go-yaml (import path `go.yaml.in/yaml/v4`).

This commit updates uses of v3 to `v4.0.0-rc.3` from the new
repository. No code changes are required.

Upgrading v2 uses to v4 will be done separately, as that requires code
changes.

Epic: none
Release note: None

#### *: upgrade yaml.v2 uses to v4

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/157866)
<!-- Reviewable:end -->

cockroach-teamcity commented:
<details><summary><strong>⚪ Sysbench</strong> [SQL, 3node, oltp_read_write]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 11.65m ±1% | 11.73m ±1% | ~ | p=0.037 n=15    |
| ⚪ **allocs/op** | 8.209k ±0% | 8.222k ±0% | ~ | p=0.118 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/e193739/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/e193739ae1c8ecceb856215ac2f502f042aabb05/bin/pkg_sql_tests benchdiff/e193739/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/e193739/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/fd29a93/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/fd29a935f4434bce5d9c2d3fa3499161c205d9c7/bin/pkg_sql_tests benchdiff/fd29a93/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/fd29a93/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/SQL/3node/oltp_read_write$ --old=fd29a93 --new=e193739 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_read_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 3.330m ±1% | 3.331m ±0% | ~ | p=0.967 n=15    |
| ⚪ **allocs/op** | 2.080k ±0% | 2.081k ±0% | ~ | p=0.218 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/e193739/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/e193739ae1c8ecceb856215ac2f502f042aabb05/bin/pkg_sql_tests benchdiff/e193739/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/e193739/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/fd29a93/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/fd29a935f4434bce5d9c2d3fa3499161c205d9c7/bin/pkg_sql_tests benchdiff/fd29a93/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/fd29a93/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_read_only$ --old=fd29a93 --new=e193739 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_write_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 3.584m ±1% | 3.591m ±0% | ~ | p=0.412 n=15    |
| ⚪ **allocs/op** | 4.191k ±0% | 4.190k ±0% | ~ | p=0.644 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/e193739/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/e193739ae1c8ecceb856215ac2f502f042aabb05/bin/pkg_sql_tests benchdiff/e193739/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/e193739/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/fd29a93/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/fd29a935f4434bce5d9c2d3fa3499161c205d9c7/bin/pkg_sql_tests benchdiff/fd29a93/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/fd29a93/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_write_only$ --old=fd29a93 --new=e193739 ./pkg/sql/tests
```

</details>

</details>

<details><summary>Artifacts</summary>

**download**:
```shell
mkdir -p new
gcloud storage cp gs://cockroach-microbench-ci/artifacts/e193739ae1c8ecceb856215ac2f502f042aabb05/19599286715-1/\* new/
mkdir -p old
gcloud storage cp gs://cockroach-microbench-ci/artifacts/fd29a935f4434bce5d9c2d3fa3499161c205d9c7/19599286715-1/\* old/
```

</details>

_built with commit: [e193739ae1c8ecceb856215ac2f502f042aabb05](https://github.com/cockroachdb/cockroach/commit/e193739ae1c8ecceb856215ac2f502f042aabb05)_

<!-- microbench-ci -->

RaduBerinde committed:
go.mod: update yaml.v4 dependency

Epic: none
Release note: None

RaduBerinde committed:
*: upgrade yaml uses to v4

The custom marshal/unmarshal interfaces differ between v2 and v3/v4,
which is a problem for inter-operability (we can't marshal/unmarshal a
struct that contains both v2 and a v3/v4 substructs).

This commit upgrades the UnmarshalYAML implementations and changes
uses of `yaml.UnmarshalStrict` (which no longer exists) to
`yamlutil.UnmarshalStrict`.

Epic: none
Release note: None

RaduBerinde committed:
cli, log: fix up tests

RaduBerinde committed:
*: use yamlutil.Marshal instead of yaml.v2.Marshal

In v3/v4, `yaml.Marshal` produces different output, which results in a
huge number of tests and testdata files changing. Use
`yamlutil.Marshal` to produce output closer to v2.

Epic: none
Release note: None

RaduBerinde committed:
docs: regenerate metrics

cockroach-teamcity commented:
<details><summary><strong>⚪ Sysbench</strong> [SQL, 3node, oltp_read_write]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 10.78m ±2% | 10.72m ±2% | ~ | p=0.389 n=15    |
| ⚪ **allocs/op** | 7.968k ±2% | 7.975k ±2% | ~ | p=0.229 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/a1ecb7b/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/a1ecb7bfdd1845148ae90c7a7a5022dff89973af/bin/pkg_sql_tests benchdiff/a1ecb7b/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/a1ecb7b/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/4f98adc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/4f98adc10870bd9edf306bfa97a79f0d49b3df1a/bin/pkg_sql_tests benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/SQL/3node/oltp_read_write$ --old=4f98adc --new=a1ecb7b ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_read_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 3.543m ±1% | 3.521m ±1% | ~ | p=0.061 n=15    |
| ⚪ **allocs/op** | 2.076k ±1% | 2.084k ±1% | ~ | p=0.717 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/a1ecb7b/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/a1ecb7bfdd1845148ae90c7a7a5022dff89973af/bin/pkg_sql_tests benchdiff/a1ecb7b/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/a1ecb7b/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/4f98adc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/4f98adc10870bd9edf306bfa97a79f0d49b3df1a/bin/pkg_sql_tests benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_read_only$ --old=4f98adc --new=a1ecb7b ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_write_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 3.844m ±1% | 3.861m ±1% | ~ | p=0.137 n=15    |
| ⚪ **allocs/op** | 4.158k ±0% | 4.174k ±0% | ~ | p=0.123 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/a1ecb7b/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/a1ecb7bfdd1845148ae90c7a7a5022dff89973af/bin/pkg_sql_tests benchdiff/a1ecb7b/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/a1ecb7b/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/4f98adc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/4f98adc10870bd9edf306bfa97a79f0d49b3df1a/bin/pkg_sql_tests benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_write_only$ --old=4f98adc --new=a1ecb7b ./pkg/sql/tests
```

</details>

</details>

<details><summary>Artifacts</summary>

**download**:
```shell
mkdir -p new
gcloud storage cp gs://cockroach-microbench-ci/artifacts/a1ecb7bfdd1845148ae90c7a7a5022dff89973af/20471760220-1/\* new/
mkdir -p old
gcloud storage cp gs://cockroach-microbench-ci/artifacts/4f98adc10870bd9edf306bfa97a79f0d49b3df1a/20471760220-1/\* old/
```

</details>

_built with commit: [a1ecb7bfdd1845148ae90c7a7a5022dff89973af](https://github.com/cockroachdb/cockroach/commit/a1ecb7bfdd1845148ae90c7a7a5022dff89973af)_

<!-- microbench-ci -->

RaduBerinde committed:
*: allow deprecated yaml calls

Cleaning these up is left for future work.

Epic: none
Release note: None

cockroach-teamcity commented:
<details><summary><strong>⚪ Sysbench</strong> [SQL, 3node, oltp_read_write]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 11.32m ±1% | 11.27m ±1% | ~ | p=0.389 n=15    |
| ⚪ **allocs/op** | 8.164k ±1% | 8.165k ±1% | ~ | p=0.862 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/905af65/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/905af65350131d817c30832f8a0ac2729b8f7db5/bin/pkg_sql_tests benchdiff/905af65/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/905af65/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/4f98adc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/4f98adc10870bd9edf306bfa97a79f0d49b3df1a/bin/pkg_sql_tests benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/SQL/3node/oltp_read_write$ --old=4f98adc --new=905af65 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_read_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 3.371m ±1% | 3.357m ±0% | ~ | p=0.037 n=15    |
| ⚪ **allocs/op** | 2.083k ±0% | 2.082k ±0% | ~ | p=0.291 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/905af65/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/905af65350131d817c30832f8a0ac2729b8f7db5/bin/pkg_sql_tests benchdiff/905af65/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/905af65/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/4f98adc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/4f98adc10870bd9edf306bfa97a79f0d49b3df1a/bin/pkg_sql_tests benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_read_only$ --old=4f98adc --new=905af65 ./pkg/sql/tests
```

</details>

</details>
<details><summary><strong>⚪ Sysbench</strong> [KV, 3node, oltp_write_only]</summary>

| Metric                      | Old Commit     | New Commit     | Delta      | Note         |
|-----------------------------|----------------|----------------|------------|--------------|
| ⚪ **sec/op** | 3.653m ±1% | 3.668m ±1% | ~ | p=0.367 n=15    |
| ⚪ **allocs/op** | 4.188k ±0% | 4.185k ±0% | ~ | p=0.288 n=15    |

<details><summary>Reproduce</summary>

**benchdiff binaries**:
```shell
mkdir -p benchdiff/905af65/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/905af65350131d817c30832f8a0ac2729b8f7db5/bin/pkg_sql_tests benchdiff/905af65/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/905af65/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
mkdir -p benchdiff/4f98adc/bin/1058449141
gcloud storage cp gs://cockroach-microbench-ci/builds/4f98adc10870bd9edf306bfa97a79f0d49b3df1a/bin/pkg_sql_tests benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
chmod +x benchdiff/4f98adc/bin/1058449141/cockroachdb_cockroach_pkg_sql_tests
```
**benchdiff command**:
```shell
benchdiff --run=^BenchmarkSysbench/KV/3node/oltp_write_only$ --old=4f98adc --new=905af65 ./pkg/sql/tests
```

</details>

</details>

<details><summary>Artifacts</summary>

**download**:
```shell
mkdir -p new
gcloud storage cp gs://cockroach-microbench-ci/artifacts/905af65350131d817c30832f8a0ac2729b8f7db5/20495680646-1/\* new/
mkdir -p old
gcloud storage cp gs://cockroach-microbench-ci/artifacts/4f98adc10870bd9edf306bfa97a79f0d49b3df1a/20495680646-1/\* old/
```

</details>

_built with commit: [905af65350131d817c30832f8a0ac2729b8f7db5](https://github.com/cockroachdb/cockroach/commit/905af65350131d817c30832f8a0ac2729b8f7db5)_

<!-- microbench-ci -->
